core/storage/btree.rs

```diff
@@ -1755,138 +1755,169 @@ impl BTreeCursor {
+                        ),
+                        self.usable_space(),
+                    )?;
 
-        let contents = page.get().contents.as_ref().unwrap();
-        let cell = contents.cell_get(
-            cell_idx,
-            self.pager.clone(),
-            payload_overflow_threshold_max(contents.page_type(), self.usable_space() as u16),
-            payload_overflow_threshold_min(contents.page_type(), self.usable_space() as u16),
-            self.usable_space(),
-        )?;
+                    if cell_idx >= contents.cell_count() {
+                        return_corrupt!(format!(
+                            "Corrupted page: cell index {} is out of bounds for page with {} cells",
+                            cell_idx,
+                            contents.cell_count()
+                        ));
+                    }
 
-        if cell_idx >= contents.cell_count() {
-            return_corrupt!(format!(
-                "Corrupted page: cell index {} is out of bounds for page with {} cells",
-                cell_idx,
-                contents.cell_count()
-            ));
-        }
+                    let original_child_pointer = match &cell {
+                        BTreeCell::TableInteriorCell(interior) => Some(interior._left_child_page),
+                        _ => None,
+                    };
 
-        let original_child_pointer = match &cell {
-            BTreeCell::TableInteriorCell(interior) => Some(interior._left_child_page),
-            _ => None,
-        };
+                    return_if_io!(self.clear_overflow_pages(&cell));
 
-        return_if_io!(self.clear_overflow_pages(&cell));
+                    let page = self.stack.top();
+                    return_if_locked!(page);
+                    if !page.is_loaded() {
+                        self.pager.load_page(page.clone())?;
+                        return Ok(CursorResult::IO);
+                    }
 
-        let page = self.stack.top();
-        return_if_locked!(page);
-        if !page.is_loaded() {
-            self.pager.load_page(page.clone())?;
-            return Ok(CursorResult::IO);
-        }
+                    page.set_dirty();
+                    self.pager.add_dirty(page.get().id);
 
-        page.set_dirty();
-        self.pager.add_dirty(page.get().id);
+                    let contents = page.get().contents.as_mut().unwrap();
 
-        let contents = page.get().contents.as_mut().unwrap();
+                    // If this is an interior node, we need to handle deletion differently
+                    // For interior nodes:
+                    // 1. Move cursor to largest entry in left subtree
+                    // 2. Copy that entry to replace the one being deleted
+                    // 3. Delete the leaf entry
+                    if !contents.is_leaf() {
+                        // 1. Move cursor to largest entry in left subtree
+                        return_if_io!(self.prev());
```

Comment:
Using `prev` to go to left child seems heavyweight and problematic if we have to make this reentrant. I think it could be fine for now until we optimize it but we need to make this a different state.